components/ProfileLessons.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import '../scss/profileLessons.scss'
+
+type Lesson = {
+  order: number
+  progress?: number
+}
+
+type LessonsProp = {
+  lessons: Lesson[]
+}
+
+const LessonImage: React.FC<Lesson> = ({ progress = 0, order }) => {
+  const lessonProgress = progress + '%'
+  const color = lessonProgress === '100%' ? 'success' : 'primary'
+  const opacity = lessonProgress === '0%' ? 0.5 : 1
+
+  return (
+    <div className="lessonImage_container" style={{ opacity: opacity }}>
+      {progress && (
+        <p
+          className={`lessonImage_progressBadge badge badge-pill badge-${color}`}
+        >
+          {lessonProgress}
+        </p>
+      )}
+      <img src={`/curriculumAssets/lessonCoversSvg/js-${order}-cover.svg`} />
+    </div>
+  )
+}
+
+const ProfileLessons: React.FC<LessonsProp> = ({ lessons }) => {
+  const displayLessons = lessons.map((lesson, i) => {
+    return (
+      <LessonImage key={i} order={lesson.order} progress={lesson.progress} />
```

Comment:
use _.uniqueId() for keys. referring to @Wolfy64 comment

> It is not recommended to use `index` as `key` with React.
> Since we use lodash you might use `_.uniqueId()` [link](https://lodash.com/docs/4.17.15#uniqueId)

Reply:
Let's not use uniqueId. We don't need it. Let's use `lesson.order` since each lesson is unique.